src/VisualStudio/Core/Def/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -118,6 +118,9 @@ internal sealed class SourceGeneratedFileManager : IOpenTextBufferEventListener
                 File.WriteAllText(temporaryFilePath, "");
             }
 
+            // Normalize file path (hint name can contain mix of slashes).
+            temporaryFilePath = Path.GetFullPath(temporaryFilePath);
```

Comment:
Although this is news to me we allowed directory support here, there's a comment that implies to me we would have normalized the hint path to the forward slash. In that case, can we just update line 111 to do document.HintName.Replace('/', '\\') rather than doing the full normalization here?

Reply:
And if it's not normalizing that way then we might need to discuss further since there's more ramifications we need to sort out.

Reply:
You're right, that's seems simpler.